### PR TITLE
[feature][KLTB002-14736] Add Jira support

### PR DESCRIPTION
**Pull request by drazenzen**
## Description
This Pull Request add Jira support to `kolibree_output`.

## Preflight Checklist

- [x] No warnings or linting issues have been introduced

##
#### Jira ticket
For more details check the [JIRA ticket](https://kolibree.atlassian.net/browse/KLTB002-14736).

#### How it works
Jira issue (summary field) is retrieved and added to changelog ouput in case Github description is missing.

### CODE REVIEW

File: src/gitchangelog/gitchangelog.py [hunk 1]
```diff
@@ -1394,6 +1421,29 @@ def kolibree_output(data, opts={}):
...
+    jira = get_jira(jira_server, jira_username, jira_apitoken)
+    RE_TICKET = None
+    if jira:
+        try:
+            jira = JIRA(
+                server=jira_server,
+                basic_auth=(jira_username, jira_apitoken)
+            )
+        except Exception:
+            die("Unable to connect to Jira")
```
Comment (georgescumihai) [inline src/gitchangelog/gitchangelog.py]: This is already handled in `get_jira`

**Comment by georgescumihai** [comment]
We should use Jira as the first choice for the changelog entry title and github as a backup in case the git commit doesn't have a jira ticket.
If jira is used, add the jira ticket url to the changelog entry for `kolibree` output engine. using the following format
`[JIRA-Ticket](jira_ticket_url) Jira ticket summary`

**Review by georgescumihai — CHANGES_REQUESTED** [review]

File: src/gitchangelog/gitchangelog.py (hunk 1 above)
Comment (drazenzen) [reply src/gitchangelog/gitchangelog.py]: Yes, that is a leftover, cleaned up.

File: src/gitchangelog/gitchangelog.py [hunk 2]
```diff
@@ -1418,10 +1460,24 @@ def render_version(version):
...
 
     def render_commit(commit, opts=opts):
-        subject = commit["subject"]
+        # Get Jira summary
+        ticket = RE_TICKET.search(commit["subject"])
+        ticket = ticket.group()[1:-1] if ticket else None
+        if ticket:
+            try:
+                issue = jira.issue(ticket, fields="summary")
+                subject = "[[{}]]({}) {}".format(
```
Comment (georgescumihai) [inline src/gitchangelog/gitchangelog.py]: there is one pair of extra brackets 
Output:
`- [[KLTB002-14238]](https://kolibree.atlassian.net/browse/KLTB002-14238) Setup repository for changelog`

Expected:
`- [KLTB002-14238](https://kolibree.atlassian.net/browse/KLTB002-14238) Setup repository for changelog`

```suggestion
                subject = "[{}]({}) {}".format(
```

**Review by georgescumihai — CHANGES_REQUESTED** [review]
Nice work, I found some small issues but should be fast to fix.

There is a small issue with the output for jira, with the extra brackets
Add `requirements.txt` for the Jira package and maybe for Github as well. if you do not install them the script will fail.

File: src/gitchangelog/gitchangelog.py (hunk 2 above)
Comment (drazenzen) [reply src/gitchangelog/gitchangelog.py]: Ok, thought that we want Jira link visually inside square brackets :) Will remove them.

**Comment by drazenzen** [comment]
> Nice work, I found some small issues but should be fast to fix.
> 
> There is a small issue with the output for jira, with the extra brackets
> Add `requirements.txt` for the Jira package and maybe for Github as well. if you do not install them the script will fail.

Added root `requirements.txt` with required libs (`PyGithub` and `jira`) and updated `setup.py` if we are going to install/use package through `setup.py`

File: src/gitchangelog/gitchangelog.py (hunk 2 above)
Comment (drazenzen) [reply src/gitchangelog/gitchangelog.py]: Updated. Lets please keep notes of any other formatting glitch/issue/... and tackle it in following formatting task.

**Review by georgescumihai — APPROVED** [review]

**Comment by lookashc** [comment]
Made tests on android monorepo, config available on branch https://github.com/kolibree-git/android-monorepo/tree/changelog
Command: `gitchangelog sdk/7.1.0..sdk/9.0.0`
Results: [sdk_9_0_0.txt](https://github.com/kolibree-git/gitchangelog/files/6042169/sdk_9_0_0.txt)
Github diff for comparison: https://github.com/kolibree-git/android-monorepo/compare/sdk/7.1.0...sdk/9.0.0

Jira integration works 🎉 

I noticed we are making the choice between sections based on `section_regexps` in config - while we should do that based on Jira ticket type, as defined here https://kolibree.atlassian.net/wiki/spaces/SOF/pages/950468787/Generator+Requirements. Currently, all Jira-related items are in `Features` section while they should be spread between different sections, based on the associated ticket type. But I guess this is for the next step - @georgescumihai can you confirm?

✅  Great job @drazenzen 🥇

**Review by lookashc — APPROVED** [review]

**Comment by georgescumihai** [comment]
Yes, there are future stories to group tickets based on Jira issue type and exclude the tickets which are not done. 
The scope of this story is only for the Jira integration.

**Pull request merged**